Scripts/GetCausalCAVIAR.py

```diff
@@ -0,0 +1,170 @@
+
+    # Load strs Regression
+    PROGRESS("\nLoad strs regression", printit=DEBUG)
+    strs = pd.read_csv(REG_STRs, sep="\t")
+    strs = strs.loc[strs['chrom']==CHROM]
+
+    # Load snps regression
+    PROGRESS("\nLoad snps regression", printit=DEBUG)
+    snps = pd.read_csv(REG_SNPs, sep="\t")
+    snps = snps.loc[snps['chrom']==CHROM]
+
+    #Load SNP genotypes
+    PROGRESS("Load SNPs", printit=DEBUG)
+    snpgt = pd.read_csv(SNPGTFILE, sep="\t")
+    snpgt = snpgt.loc[snpgt['chrom']==CHROM]
+
+    # Load STR genotypes
+    PROGRESS("Load STRs", printit=DEBUG)
+    strgt = pd.read_csv(STRGTFILE, sep="\t")
+    strgt = strgt.loc[strgt['chrom']==CHROM]
+
+    # Restrict to STR samples
+    PROGRESS("Restrict to STRs samples", printit=DEBUG)
+    str_samples = list(set(strgt.columns[2:].values).intersection(set(snpgt.columns[2:].values)))
+    expr = expr.loc[str_samples,:]
+    snpgt = snpgt[["chrom","start"] + str_samples]
+    snpgt.index = list(snpgt["start"].apply(lambda x: "SNP_%s"%int(x)))
+    strgt = strgt[["chrom","start"] + str_samples]
+    strgt.index = list(strgt["start"].apply(lambda x: "STR_%s"%int(x)))
+    samples_to_keep = str_samples
+
+    # Load eSTR results
+    PROGRESS("Restrict to eSTR genes only", printit=DEBUG)
+    if ESTRGENESFILE is not None:
+        estr_genes = pd.read_csv(ESTRGENESFILE, sep="\t")
+        Genes = estr_genes.loc[estr_genes['qvalue']<=0.1]['gene']  # estrs at 10%FDR
+        expr_annot = expr_annot.loc[expr_annot['gene.id'].isin(list(Genes))]
+
+# For each gene, get all cis-variants and the best STR
+    for i in range(expr_annot.shape[0]):
+        gene=expr_annot.index.values[i]
+        ensgene = expr_annot["gene.id"].values[i]
+        genedir=TMPDIR+"/%s"%gene
+        if not os.path.exists(genedir):
+            os.mkdir(genedir)
+        PROGRESS("Getting data for %s"%gene, printit=DEBUG)
+        start = expr_annot["gene.start"].values[i]
+        end = expr_annot["gene.stop"].values[i]
+    # Pull out cis SNPs
+        PROGRESS("Getting cis SNPs for %s"%gene)
+        cis_snps = snps[(snps["str.start"] >= (start-DISTFROMGENE)) & (snps["str.start"] <= (end+DISTFROMGENE))]
+        cis_snps = cis_snps.loc[cis_snps['gene']==ensgene]
+        cis_snps.index = cis_snps["str.start"].apply(lambda x: "SNP_%s"%int(x))
+        L=list(cis_snps.index)
+    # Pull out most significant STR
+        PROGRESS("Getting most significant cis STR for %s"%gene)
+        best_str_start = strs[strs["gene"]==ensgene].sort("p.wald")["str.start"].values[0]
+        cis_strs = strs.loc[strs['gene']==ensgene]
+        cis_strs.index = list(cis_strs['str.id'])
+        try:
+            del cis_strs['ID']
+        except:
+            pass
+        cis_snps.loc['STR_'+str(best_str_start)] = list(cis_strs.loc['STR_'+str(best_str_start)])
+    # Make z file data
+        Z = z(cis_snps[['beta','beta.se']])
+        Z.to_csv(genedir+'/ZFILE', sep='\t',header=None)
```

Comment:
I would rename the function ```z``` to something longer/more informative. It is too easy to accidentally reassign something to a single letter variable name.